### PR TITLE
fix: stop time incrementer on workspace page

### LINKED ISSUE #5354: Schedule bumper on WorkspacePage can't increase time
Decreasing time works. Increasing time causes "new deadline must be at least 30 minutes into the future" error even when the new deadline is far enough into the future.

### PR DESCRIPTION
**Pull request by presleyp**
Fixes #5354 

The incrementer wasn't working when the template default ttl was 0, because the max deadline was being calculated as start time + 0.

### CODE REVIEW

**Review by BrunoQuaresma — APPROVED** [review]
LGTM, would make sense for us to create a test for this?

**Comment by presleyp** [comment]
@BrunoQuaresma yes, but when I went to I realized the max deadline doesn't depend on the template at all anymore since the template has a default rather than a max - reworking it now!

**Comment by presleyp** [comment]
I have tests in mind for this but I'm having some trouble with them, so in the interest of wrapping up PRs before the break I'd like to merge this and come back for the tests in #5476.

File: site/src/components/WorkspaceScheduleButton/EditHours.tsx [hunk 1]
```diff
@@ -19,13 +19,13 @@ export const EditHours = ({
   const styles = useStyles()
 
   return (
-    <form onSubmit={() => handleSubmit(hours)}>
+    <form onSubmit={() => handleSubmit(Number.isNaN(hours) ? 0 : hours)}>
```
Comment (BrunoQuaresma) [inline site/src/components/WorkspaceScheduleButton/EditHours.tsx:22]: when is it going to be `NaN`?

Comment (presleyp) [reply site/src/components/WorkspaceScheduleButton/EditHours.tsx:22]: When you delete the number from the input

**Review by BrunoQuaresma — APPROVED** [review]
@presleyp that sounds fair enough. I just had a minor question.

File: site/src/components/WorkspaceScheduleButton/EditHours.tsx (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/components/WorkspaceScheduleButton/EditHours.tsx:22]: Maybe we could add a comment explaining this? So we know why we are checking it. Up to you.

Comment (presleyp) [reply site/src/components/WorkspaceScheduleButton/EditHours.tsx:22]: Good idea!

**Pull request merged**